### ISSUE TITLE
Latest version is segfaulting on Ubuntu 14.04, CUDA 7.5

### ISSUE THREAD

**Issue by leeping** [pandegroup/openmm#1472]
Title says it all.  :)  After I reverted back to commit 8190f24389d8c025f3dc99c99bd94c961c3114d9 (when Peter implemented AMOEBA periodic bonds), it worked again.

**Comment by jchodera** [comment]
Have you tried the `conda` package?  Or are you compiling from source?

**Comment by leeping** [comment]
Sorry for the slow response.  I'm compiling from source.

**Comment by leeping** [comment]
Yudong Qiu, the postdoc in my lab (@yudongqiu) looked at the commit history, and it seems that 25d214540bc9191478f13e5f0a4f549fd96b0a96 is the latest one that works, and only the most recent commit (9e66f0b1ec1fbc4d8dfe991dc67b6be1190d0f08) leads to the segfault.

**Comment by peastman** [comment]
What exactly is producing segfaults?  I just tried some test cases on Ubuntu 14.04 with CUDA 7.5 and didn't see any.

**Comment by yudongqiu** [comment]
I found that the testcases all passed but the HelloArgon and testInstallation.py both reported the malloc error:
./HelloArgon
HelloArgon: malloc.c:2372: sysmalloc: Assertion `(old_top == (((mbinptr) (((char *) &((av)->bins[((1) - 1) * 2])) - __builtin_offsetof (struct malloc_chunk, fd)))) && old_size == 0) || ((unsigned long) (old_size) >= (unsigned long)((((__builtin_offsetof (struct malloc_chunk, fd_nextsize))+((2 *(sizeof(size_t))) - 1)) & ~((2 *(sizeof(size_t))) - 1))) && ((old_top)->size & 0x1) && ((unsigned long) old_end & pagemask) == 0)' failed.
Aborted
This only happens after the lastest commit (merged pull request #1469).
Ubuntu 16.04 LTS, CUDA 7.5 has the same problem.

**Comment by peastman** [comment]
I just ran HelloArgon, and it ran without problem.  Could you run it in gcc, then type "bt" to get a stack trace of where the segfault is happening?

You might also try deleting your build directory, than recreating it and rebuilding from scratch.  It may be that something in it has just gotten out of sync.

Also, what compiler are you using?

**Comment by mpharrigan** [comment]
run it in `gdb`, you mean

**Comment by yudongqiu** [comment]
I was using gcc and g++ and I was building from scratch.
From the backtrace I realized the problem. We had a environmental variable OPENMM_PLUGIN_DIR set to a directory that contains libOpenMMCPU.so which was from previous versions. Although we built and installed the latest version, HelloArgon and other openmm programs still use the old one for functions like registerPlatforms(). After I unset that environmental variable and rebuilt, there is no problem now.
Thanks very much for the suggestion of backtracing since ldd was not showing the old libray file.
